src/CTA.WebForms2Blazor/ControlConverters/ControlConverter.cs

```diff
@@ -64,10 +113,48 @@ public static HtmlDocument ConvertEmbeddedCode(HtmlDocument document)
 
             //Not implemented/used yet
             //htmlString = EmbeddedCodeReplacers.AspExpRegex.Replace(htmlString, aspExprEval);
+            
+            return htmlString;
+        }
+        
+        //This function only updates the first child node that matches the name,
+        //but for current purposes there should only be one matching node
+        protected bool UpdateInnerHtmlNode(HtmlNode outerNode, string targetName, 
+            string template = null, 
+            string newName = null, 
+            IEnumerable<Attribute> newAttributes = null, 
+            string newBody = null)
+        {
+            var lowerName = targetName.ToLower();
+
+            //Ideally use .SelectSingleNode, but doesnt work with names with ':' character and is less flexible
+            //var selectedNode = node.SelectSingleNode(lowerName);
+            
+            var selectedNodes = outerNode.Descendants().Where(child =>
+            {
+                return child.Name.ToLower() == lowerName;
+            }).ToList();
+
+            if (selectedNodes.Count > 0)
+            {
+                template??= NodeTemplate;
+                newName??= targetName;
```

Comment:
template had to be done this way since NodeTemplate is not a compile time constant, and I couldn't figure out a better way to default assign newName to be equal to another argument value